File: pulse-cpp/src/test/cpp/evaluationtest.cpp
// Copyright 2013-2021 Phokham Nonava
//
// Use of this source code is governed by the MIT license that can be
// found in the LICENSE file.

#include "evaluation.h"
#include "notation.h"

#include "gtest/gtest.h"

using namespace pulse;

TEST(evaluationtest, testEvaluate) {
	Position position(notation::toPosition(notation::STANDARDPOSITION));

	EXPECT_EQ(+evaluation::TEMPO, evaluation::evaluate(position));
}

File: pulse-java/src/test/java/com/fluxchess/pulse/java/EvaluationTest.java
/*
 * Copyright 2013-2023 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse.java;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class EvaluationTest {

	@Test
	void testEvaluate() {
		Position position = Notation.toPosition(Notation.STANDARDPOSITION);
		Evaluation evaluation = new Evaluation();

		assertThat(evaluation.evaluate(position)).isEqualTo(Evaluation.TEMPO);
	}
}
